Library/Homebrew/extend/ENV/shared.rb

```diff
@@ -190,10 +190,16 @@ def ncurses_define
 
   # @private
   def userpaths!
-    paths = ORIGINAL_PATHS.map { |p| p.realpath.to_s rescue nil } - %w[/usr/X11/bin /opt/X11/bin]
-    self["PATH"] = paths.unshift(*self["PATH"].split(File::PATH_SEPARATOR)).uniq.join(File::PATH_SEPARATOR)
+    paths = self["PATH"].split(File::PATH_SEPARATOR)
+    # put Superenv.bin and opt path at the first
+    new_paths = paths.select { |p| p.start_with?("#{HOMEBREW_REPOSITORY}/Library/ENV") || p.start_with?("#{HOMEBREW_PREFIX}/opt") }
     # XXX hot fix to prefer brewed stuff (e.g. python) over /usr/bin.
-    prepend_path "PATH", HOMEBREW_PREFIX/"bin"
+    new_paths << "#{HOMEBREW_PREFIX}/bin"
```

Comment:
Is this still necessary? If the Homebrew dependencies we collected are already at the front of `PATH`, we should respect the ordering of `ORIGINAL_PATHS` for anything else.